webpack.mix.js

```diff
@@ -19,6 +19,7 @@ mix.setPublicPath('src/public');
 
 mix.js('src/resources/assets/js/geo/main.js', 'assets/scripts').vue()
     .js('src/resources/assets/js/volumes/volumes.js', 'assets/scripts').vue()
+    .js('src/resources/assets/js/annotations/annotations.js', 'assets/scripts').vue()
```

Comment:
Hi @mzur,
I was not entirely sure how I could add / inject an `annotationsSettingsTab`-plugin into the BIIGLE annotation view, so I manually added this line in the `webpack.mix.js`-file in order to publish the annotations.js script that eventually defines the plugin. However, the [.vue plugin component](https://github.com/biigle/geo/pull/60/files#diff-94f4a070c8c000dbc91671c354a3b1906519c22d3b803e9917730f9b3c4c3e4e) seems not being rendered, as I get the warning message that it's properties are _"...not defined on the instance but referenced during render."_

Therefore, I reason that manually publishing the annotations.js is not the right way to implement the plugin. Could you provide some help to correctly implement a plugin?

Reply:
Adding the line to webpack.mix.js will produce a separate file for your code. To make it work in the annotation tool, you have to load the file through the appropriate view mixin, too, which is `annotationsScripts` ([example](https://github.com/biigle/reports/blob/master/src/resources/views/annotationsScripts.blade.php)).

Reply:
Yes, I did have a look at the example you provided earlier and tried that [here](https://github.com/biigle/geo/pull/60/files#diff-7845294668f749758a8942a09d7cbb7e5a97e4ed470c58172cfe5aacef25baf3), but it doesn't seem to make a difference.
I also ran `php artisan geo:publish` to publish the annotations script.

Reply:
Well... I forgot a closing `</script>` tag in the .vue plugin component...